### ISSUE TITLE
New Region Names (8-point scale) for described proximity to ratio

### ISSUE THREAD

**Issue by terracoda** [phetsims/ratio-and-proportion#137]
@zepumph, we have a new 9-point scale for describing the proximity to ratio. Hopefully, this will match nicely with the 9-point scale for describing position. 

They are inspired from Gravity Force Lab Basics.

I have no rough ranges for these described ratios, hopefully, you have some ideas on how to handle this dynamically based on the model values.

Here are the parameters:
- farthest from
- extremely far from
- very far from
- far from
- not so close to / somewhat close to
- close to
- very close to
- extremely close to
- at

These parameters are followed by "challenge ratio" or "my challenge ratio" in the dynamic descriptions.

I am not yet sure what the responses will say.

**Comment by terracoda** [comment]
@zepumph, just an FYI, I am working on how these scales will be used in the design doc.

**Comment by zepumph** [comment]
pedagogically, I feel like "farthest from" is a bit of a confusing  region for this sim. It doesn't have an obvious mapping in the code, and I don't know how helpful it would be. If we want it, I see a couple of possible ways to proceed:

* Mapping from fitness of 0-1 (0 is when background is white). So 0 would be "farthest from" this is weird to me because the ratio can change and still be "farthest from".
* We can calculate the maximum incorrect ratio given a target ratio, it will likely be  when one hand is at max, and one hand is at min. That said, it will likely always be left at max, and right at min. Then we just linearly or piecewise map the middle regions (like we have done before in other sims) 

I think I would most rather exclude "fathest from" so that we can choose an arbirary point that is "quite far" and call everything farther than that "extremely far from"

Let me know what you would like

**Comment by terracoda** [comment]
Ok, maybe, "farthest from other hand" makes more sense, as a kind of landmark. This would be similar to how we use "Track end left" and "Track end right" in GFLB.

Can you just merge the single-value region into "extremely far from" for now. We may only need an 8-point scale.

I hope to get some examples of the responses done in the morning.

**Comment by zepumph** [comment]
>Ok, maybe, "farthest from other hand" makes more sense, as a kind of landmark. This would be similar to how we use "Track end left" and "Track end right" in GFLB.

I think a single value region like this seems weird to me because of how it is relational based on both hand values. I'm happy to play around with it more though.

**Comment by zepumph** [comment]
I'm not going to get to this until at least Monday.

**Comment by terracoda** [comment]
Sorry for thinking out loud in a confusing way. I did not mean implement “farthest from other hand”, I am still thinking through that idea and how it would work, if at all.

I meant just make “extremely far from“ the last region for proximity to challenge ratio.

**Comment by zepumph** [comment]
Ahh, I see. Gotcha!

**Comment by terracoda** [comment]
@zepumph, can you implement 8 regions, please:

- very far from
- far from
- not so close to 
- close to
- very close to
- extremely close to
- at

Note, I chose "not so close to" over "somewhat close to", but after implementation we may need to swap that one.

**Comment by terracoda** [comment]
Just added "extremely far from" to above list.

**Comment by emily-phet** [comment]
Discussed today, and here's the final list (8-point scale) we'd like implemented:
- "extremely far from" - (screen white)
- "very far from" - (screen white)
- "far from" - (screen white)
- "not so close to" - (screen white)
- "somewhat close to" - (screen light green)
- "very close to" - (screen light green)
- "extremely close to" - (screen light green)
- "at" - (screen darkes green)

Note - I added next to each what color the screen background would be, to help you understand how these regions must scale with the color/sound feedback provided in other modalities.

**Comment by zepumph** [comment]
Use this table: https://docs.google.com/document/d/1HJvvGWYaUv9tlat9S7omQQVng8fGB4FuUhTEM1qx1z8/edit#heading=h.v24owms95lch

**Comment by zepumph** [comment]
A reminder for me, well explained above, that the `at` region occupies the entire range of `model.inProportion()`

**Comment by zepumph** [comment]
>A reminder for me, well explained above, that the at region occupies the entire range of model.inProportion()

This is already in the describer code:

```js
    if ( this.model.inProportion() ) {
      return RATIO_FITNESS_STRINGS_CAPITALIZED.length - 1; // "at"
    }
```

I'm a bit worried about how to define the regions that exist when fitness is 0. A couple of ideas:
* base it on distances in ratio error from the target ratio
* build it into the model with a new value range simliar to fitness, but with a different algorithm. This way other modalities to monitor this new range.
* Calculate the min and max ratio possible, and linearly interpolate between the current ratio. Most likely this will yield inaccurate ranges, since a linear scale will not reflect the multiplicative relationships the positions have on the ratio.
* I think first I will try to just get something built into the sim so that we can try it out.

**Comment by zepumph** [comment]
Ah!! An idea. What if we stop clamping fitness, at least for this context, then we have the same algorithm that we are using for fitness, just sometimes we interpret it from 0-1 (normal case), and sometimes, like for this case, we interpret it from (~-10000, 1]. I will look into it.

**Comment by zepumph** [comment]
Here is the value of unclamped fitness given the current ratio. I see the min fitness is >-50. Maybe that can be our bound for interpolation, for now. 


| Target ratio | fitness <br>at max ratio | fitness <br>at min ratio|
| -------------- | --------- | ---------- |
| 1/2 | -9 | -4|
| 1/10 | -49 | -4|
| 1/1 | -4 | -4|
| 2/1 | -4 | -9|
| 10/1 | -4 | -49|

**Comment by zepumph** [comment]
The penultimate commit above added most of the logic above, including an `unclampedRatioProperty`, This still maxes out at 1, but an go below 0 depending on how bad the fitness is. I felt like it was important to get you something to test quickly, but I think that if we move forward with this solution, then both Properties should be combined into one.

Here is the best way to test:
* on dev version: https://phet-dev.colorado.edu/html/ratio-and-proportion/1.0.0-dev.54/phet/
* run with `?log` which now displays  the unclamped fitness ranging ~(-50,1]
* Play around with it to get a feel of the range.
* in the a11y view: use the shift key navigation from startup while looking at the unclamped fitness value to note how the ranges change:
  * when in proportion: ratio is "at"
  * between 0 and "in proportion", regions map linearly from "somewhat close to" to "extremely close to"
  * between -3 and 0, regions map linearly from "extremely far from" to "not close to"
  * anything less than -3 will also map to "extremely far from".


This is a complicated review, but I was kinda going with my gut in terms of region spacing. Likely some tweaking should be made. Let me know if you would like to review together to make it simpler.

**Comment by zepumph** [comment]
After discussion today with @emily-phet, @Matthew-Moore240, and me, we would like to calculate what the current min-fitness is based on the target ratio, and map linearly from that number to 0 for the farther 4 regions. This is better than an arbitrary constant (currently -3) in which everything lower is clamped.

**Comment by zepumph** [comment]
I have based the min of the ratio fitness regions on the actual minimum possible fitness given a target ratio. @emily-phet and @Matthew-Moore240 please review and compare the regions to https://phet-dev.colorado.edu/html/ratio-and-proportion/1.0.0-dev.54/phet/ which just uses an arbitrary cutoff of a fitness of -3 to map regions less than 0.

Thanks!

**Comment by terracoda** [comment]
@zepumph, one region name string on master is missing "so".

- "not close to challenge ratio"

Should be:
- "not so close to challenge ratio"

I see the "so" in the distance region name, but not in the proximity to ratio region name.  

Also, unrelated to this issue, in this view (with tick marks), I had proposed using number of tick marks apart as the metric for describing distance. I'll check in with @Matthew-Moore240 on that, and start a new issue if necessary.

**Comment by terracoda** [comment]
I am re-assigning back to @zepumph. I hope this is okay?

**Comment by terracoda** [comment]
Also, it seems odd to me that with left hand at zero and right hand at half-way past zero (basically tick mark 0.5, regardless of the set tick mark range 10, 20 or 30) I can't get the ding for the 1:2 challenge ratio.

But maybe that is mathematically correct? Maybe 0 to 0.5 is not equivalent to 1:2? And maybe that is why we need a zero mark?

Here's a screenshot of my confusion:
![Screen Shot 2020-09-01 at 11 50 07 AM](https://user-images.githubusercontent.com/1902116/91864215-86b58700-ec4a-11ea-9fc4-a96857dde23b.png)

**Comment by zepumph** [comment]
>Also, it seems odd to me that with left hand at zero and right hand at half-way past zero (basically tick mark 0.5, regardless of the set tick mark range 10, 20 or 30) I can't get the ding for the 1:2 challenge ratio.

Yes, zero is weird, that is expected though

>I see the "so" in the distance region name, but not in the proximity to ratio region name.

I updated the region. Anything else for this issue @terracoda, @Matthew-Moore240?

**Comment by terracoda** [comment]
That's the change that I saw. Looks good to me @zepumph.

**Comment by terracoda** [comment]
@Matthew-Moore240, I'll go ahead and close this issue. The ratio regions are implemented.
